### PR TITLE
Fix gc hole in Frozen.cs

### LINKED ISSUE #76856: Test failure: GC\\API\\Frozen\\Frozen\\Frozen.cmd
win-arm64 random JitStress

```
set COMPlus_TieredCompilation=0
set COMPlus_JitStress=1cf

    GC\API\Frozen\Frozen\Frozen.cmd [FAIL]
      Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
         at HelloFrozenSegment.Program.Main(String[] args)
      
      Return code:      1
```

https://dev.azure.com/dnceng-public/public/_build/results?buildId=46805&view=ms.vss-test-web.build-test-results-tab&runId=941878&resultId=109785&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab

@dotnet/jit-contrib

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/issues/76856

Confirmed that the test stopped crashing after this change.

### CODE REVIEW

**Review by EgorBo — COMMENTED** [review]
-

File: src/tests/GC/API/Frozen/Frozen.cs [hunk 1]
```diff
@@ -134,16 +134,17 @@ internal class Node
...
 
+        internal static ref byte GetRawData(this object obj) =>
+            ref Unsafe.As<RawData>(obj).Data;
+
+        internal static unsafe IntPtr GetMethodTablePointer(object obj) =>
+            (IntPtr)Unsafe.Add(ref Unsafe.As<byte, IntPtr>(ref obj.GetRawData()), -1);
+
         private static unsafe int Main(string[] args)
         {
             Node template = new Node();
```
Comment (jkotas) [inline src/tests/GC/API/Frozen/Frozen.cs:150]:
```suggestion
            IntPtr methodTable = typeof(Node).TypeHandle.Value;
```

File: src/tests/GC/API/Frozen/Frozen.cs [hunk 2]
```diff
@@ -134,16 +134,17 @@ internal class Node
...
-            gch.Free();
-            return methodTable;
+            public byte Data;
         }
 
+        internal static ref byte GetRawData(this object obj) =>
+            ref Unsafe.As<RawData>(obj).Data;
+
+        internal static unsafe IntPtr GetMethodTablePointer(object obj) =>
+            (IntPtr)Unsafe.Add(ref Unsafe.As<byte, IntPtr>(ref obj.GetRawData()), -1);
```
Comment (jkotas) [inline src/tests/GC/API/Frozen/Frozen.cs:146]: This assumes that offset 0 is a valid byref. It happens to be true in CoreCLR today, but it is not a safe assumption to make.

File: src/tests/GC/API/Frozen/Frozen.cs (hunk 1 above)
Comment (jkotas) [reply src/tests/GC/API/Frozen/Frozen.cs:150]: This is much simpler and depends on the CoreCLR internals about as much as alternatives.

**Review by jkotas — APPROVED** [review]

**Review by cshung — APPROVED** [review]
Thanks for the fix

**Pull request merged**